File: BankManager_Java/DeTai7/src/main/java/com/nhk/detai7/QuanLyKhachHang.java
package com.nhk.detai7;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author 84355
 */
public class QuanLyKhachHang {
    private List<KhachHang> ds;

    public QuanLyKhachHang() {
        this.ds = new ArrayList<>();
    }

    public void themKhachHang(KhachHang... kh) {
        this.ds.addAll(Arrays.asList(kh));
    }

    public boolean isKhachHangDaCo(String cccd) {
        for (KhachHang khachHang : ds) {
            if (khachHang.getSoCCCD().equals(cccd))
                return true;
        }
        return false;
    }

    public void XuatDSKhachHang() {
        this.ds.forEach(s -> {
            System.out.printf("%s" +
                    "* Hoat dong: %s\n" +
                    "* * * * * * * * * * * * * * * * * *\n", s, s.isActive());
        });
    }

    public KhachHang dangNhap(String username, String password) {
        for (KhachHang kh : this.ds) {
            if (kh.getMaKH().equals(username) && kh.getPassword().equals(password) && kh.isActive() == true ||
                    kh.getSdt().equals(username) && kh.getPassword().equals(password) && kh.isActive() == true) {
                return kh;
            }
        }
        return null;
    }

    public KhachHang layKhachHang(String ten, String stk) {
        for (KhachHang kh : ds) {
            if (kh.getMaKH().equals(stk) && kh.getHoTen().equals(ten))
                return kh;
        }
        return null;
    }

    public KhachHang layKhachHang(String cccd) {
        for (KhachHang kh : ds) {
            if (kh.getSoCCCD().equals(cccd))
                return kh;
        }
        return null;
    }

    public boolean isSDTTonTai(String sdt) {
        for (KhachHang kh : ds) {
            if (kh.getSdt().equals(sdt))
                return true;
        }
        return false;
    }

    // ===========Getter, Setter===============
    /**
     * @return the ds
     */
    public List<KhachHang> getDs() {
        return ds;
    }

    /**
     * @param ds the ds to set
     */
    public void setDs(List<KhachHang> ds) {
        this.ds = ds;
    }
}

File: BankManager/BankManager/Lop/QuanLyKhachHang.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BankManager.Lop
{
    class QuanLyKhachHang
    {
        private List<KhachHang> ds;

        public QuanLyKhachHang()
        {
            this.ds = new List<KhachHang>();
        }

        public void ThemKhachHang(params KhachHang[] kh)
        {
            this.ds.AddRange(kh);
        }

        //public bool IsKhachHangDaCo(string cccd)
        //{
        //    foreach (KhachHang khachHang in ds)
        //    {
        //        if (khachHang.SoCCCD.Equals(cccd))
        //            return true;
        //    }
        //    return false;
        //}

        //public void XuatDSKhachHang()
        //{
        //    foreach (KhachHang kh in ds)
        //    {
        //        Console.WriteLine($"{kh}\n* Hoat dong: {kh.IsActive}\n* * * * * * * * * * * * * * * * * *");
        //    }
        //}

        //public KhachHang DangNhap(string username, string password)
        //{
        //    foreach (KhachHang kh in this.ds)
        //    {
        //        if (kh.Sdt != null && kh.Password != null && kh.Sdt.Equals(username) && kh.Password.Equals(password))
        //        {
        //            return kh;
        //        }
        //    }
        //    return null;
        //}

        //public KhachHang LayKhachHang(string ten, string stk)
        //{
        //    foreach (KhachHang kh in ds)
        //    {
        //        if (kh.MaKH.Equals(stk) && kh.HoTen.Equals(ten))
        //            return kh;
        //    }
        //    return null;
        //}

        //public KhachHang LayKhachHang(string cccd)
        //{
        //    foreach (KhachHang kh in ds)
        //    {
        //        if (kh.SoCCCD.Equals(cccd))
        //            return kh;
        //    }
        //    return null;
        //}

        //public bool IsSDTTonTai(string sdt)
        //{
        //    foreach (KhachHang kh in ds)
        //    {
        //        if (kh.Sdt.Equals(sdt))
        //            return true;
        //    }
        //    return false;
        //}

        // Getter, Setter
        public List<KhachHang> Ds
        {
            get { return this.ds; }
            set { this.ds = value; }
        }
    }
}
